### PR TITLE
use toLocaleString where available in number formatter

### PR DESCRIPTION
**Pull request by prehnRA**

### CODE REVIEW

File: app/assets/javascripts/revelry/ui/formatting/NumberFormatter.js.cjsx [hunk 1]
```diff
@@ -1,9 +1,28 @@
 Rev.registerComponent 'NumberFormatter',
+  propTypes: ->
+    value: React.PropTypes.number
+    locales: React.PropTypes.string
+    localeMatcher: React.PropTypes.oneOf ['lookup', 'best fit']
+    style: React.PropTypes.oneOf ['decimal', 'currency', 'percent']
+    currency: React.PropTypes.string
+    currencyDisplay: React.PropTypes.oneOf ['symbol', 'code', 'name']
```
Comment (jwietelmann) [inline app/assets/javascripts/revelry/ui/formatting/NumberFormatter.js.cjsx]: Does this really belong here?

Comment (prehnRA) [inline app/assets/javascripts/revelry/ui/formatting/NumberFormatter.js.cjsx]: Yes. There are only certain valid props, and they are documented here.

Comment (prehnRA) [inline app/assets/javascripts/revelry/ui/formatting/NumberFormatter.js.cjsx]: Also I can look here at a glance and figure out what valid parameters for this component are.

Comment (jwietelmann) [inline app/assets/javascripts/revelry/ui/formatting/NumberFormatter.js.cjsx]: I mean specifically currency and currencyDisplay.

Comment (prehnRA) [inline app/assets/javascripts/revelry/ui/formatting/NumberFormatter.js.cjsx]: Yes, those are parameters to toLocaleString and can be passed to this component.

Comment (prehnRA) [inline app/assets/javascripts/revelry/ui/formatting/NumberFormatter.js.cjsx]: Basically: this component does everything that toLocaleString does, and I want to stick to the rule that all valid props are documented in propTypes.

**Pull request merged**